df_engine_extensions/src/dist_sql_query/physical_plan.rs

```diff
@@ -169,19 +182,148 @@ impl<R: RemotePhysicalPlanExecutor> ExecutionPlan for ResolvedPartitionedScan<R>
+        )?;
+        let record_schema = RecordSchema::try_from(plan.schema()).map_err(|e| {
+            DataFusionError::Internal(format!(
+                "failed to convert arrow_schema to record_schema, arrow_schema:{}, err:{e}",
+                plan.schema()
+            ))
+        })?;
+        let encoded_plan = EncodedPlan {
+            plan: plan_bytes,
+            schema: record_schema,
+        };
+
+        // Send plan for remote execution.
+        let stream_future =
+            self.remote_executor
+                .execute(sub_table.clone(), &context, encoded_plan)?;
+        let record_stream = PartitionedScanStream::new(stream_future, plan.schema());
+
+        Ok(Box::pin(record_stream))
     }
 
     fn statistics(&self) -> Statistics {
         Statistics::default()
     }
 }
 
+/// Partitioned scan stream
+pub struct PartitionedScanStream {
+    /// Future to init the stream
+    stream_future: BoxFuture<'static, DfResult<DfSendableRecordBatchStream>>,
+
+    /// Stream to poll the records
+    stream_state: StreamState,
+
+    /// Record schema
+    arrow_record_schema: ArrowSchemaRef,
+}
+
+impl PartitionedScanStream {
+    /// Create an empty RecordBatchStream
+    pub fn new(
+        stream_future: BoxFuture<'static, DfResult<DfSendableRecordBatchStream>>,
+        arrow_record_schema: ArrowSchemaRef,
+    ) -> Self {
+        Self {
+            stream_future,
+            stream_state: StreamState::Initializing,
+            arrow_record_schema,
+        }
+    }
+}
+
+impl RecordBatchStream for PartitionedScanStream {
+    fn schema(&self) -> ArrowSchemaRef {
+        self.arrow_record_schema.clone()
+    }
+}
+
+impl Stream for PartitionedScanStream {
+    type Item = DfResult<RecordBatch>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = self.get_mut();
+
+        loop {
+            let stream_state = &mut this.stream_state;
+            match stream_state {
+                StreamState::Initializing => {
+                    let poll_res = this.stream_future.poll_unpin(cx);
+                    match poll_res {
+                        Poll::Ready(Ok(stream)) => {
+                            *stream_state = StreamState::Polling(stream);
+                        }
+                        Poll::Ready(Err(e)) => {
+                            *stream_state = StreamState::Failed(e.to_string());
+                            return Poll::Ready(Some(Err(e)));
+                        }
+                        Poll::Pending => return Poll::Pending,
+                    }
+                }
+
+                StreamState::Polling(stream) => {
+                    let poll_res = stream.poll_next_unpin(cx);
+                    if let Poll::Ready(Some(Err(e))) = &poll_res {
+                        *stream_state = StreamState::Failed(e.to_string());
+                    }
+
+                    return poll_res;
```

Comment:
Just return the inner stream's polling result?
```suggestion
                    return stream.poll_next_unpin(cx);
```